producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+= Using the Kafka CDC Producer
+
+This extension provides support for sending Change Data Capture(CDC) events to an Apache Kafka®.
+It is designed to work with the DataStax Stargate.
+
+== Kafka CDC - Stargate integration
+
+This extension is designed to integrate with Stargate. It uses a link:https://www.osgi.org/[OSGi] framework for registering the extension within the Stargate.
+The `KafkaProducerActivator` class handles the registration process. When you will put the `producer-kafka-{version}.jar` in the
+`/stargate-lib` directory it will be automatically picked, and the activator logic is invoked.
+
+This extension has one service required: `Metrics` - this is provided by the `CoreActivator` in the link:../core[core] module.
+
+== Architecture
+
+The main entry point for this CDC connector is a link:../producer-kafka/src/main/java/io/stargate/producer/kafka/KafkaCDCProducer.java[KafkaCDCProducer].
+It implements the SchemaAwareCDCProducer interface that serves as an integration layer between Stargate's persistence and CDC extension.
+
+1. `init()` - takes the map of configuration settings that are passed from the Stargate to CDC connector. The configuration settings are described in <<Configuration>>
+
+2. `createTableSchemaAsync()` - MUST be invoked on every Stargate node before any call to the `send` is made (including restarts).
+This method is responsible for constructing the Avro schema based on the `TableMetadata` and registering the schema in link:https://docs.confluent.io/current/schema-registry/index.html[schema registry].
+It is also invoked if any schema change on the CDC-enabled table is made. The detailed process of schema creation is described in <<Schema Creation>>
+
+3. `send()` - it is responsible for creating a link:https://avro.apache.org/docs/1.10.0/api/java/org/apache/avro/generic/GenericRecord.html[GenericRecord] based on the Avro schema created in the previous step.
+Once the record is created, it is sent to the Apache Kafka topic.
+
+4. `close()` - it flushes all pending records from the underlying Kafka producer and closes it.
+
+image::documentation/Kafka_CDC_architecture.png[Kafka CDC architecture]
+
+Let's recap the basics of architecture. The CDC connector is working within a Stargate node. The stargate node sends the TableMetadata when the schema is changed, OR the Stargate node is restarted.
+It constructs the schema and registers it in the schema registry. When the new CDC event occurs, the Stargate is propagating
+this change to the CDC extension via the send() method. The send() retrieves the schema form schema registry (or a local cache), constructs the Avro record, and sends it to Kafka.
+
+== Configuration
+
+The `init()` method takes the `Map<String, Object>` as an argument. This map must contain the required settings for running the Kafka CDC connector.
```

Comment:
How are these being provided to stargate? As system args? From a table?

Reply:
Using a new config store mechanism that does not exist yet. I plan to work on it next

Reply:
Got it 👍